packages/react/src/components/UIShell/Link.tsx

```diff
@@ -6,22 +6,49 @@
  */
 
 import PropTypes from 'prop-types';
-import React from 'react';
+import React, {
+  type ElementType,
+  type ForwardedRef,
+  forwardRef,
+  type WeakValidationMap,
+  type ComponentProps,
+  Ref,
+} from 'react';
+
+// Note: Maybe we should use `as` instead of `element`? `as` appears to be
+// standard and is used in other places in this project.
```

Comment:
@tay1orjones What do you think about switching to `as`? I could add a comment about `element` being deprecated and add `as` to the prop type

Reply:
@mrkjdy I'm not opposed to adding `as`, but we'll need to keep `element` around. We have a deprecation warning helper that can be placed on `element` proptypes. The `@deprecated` ts annotation can be added as well. We'll also need to map legacy usages of `element` to be the value  for `as` so nothing breaks for folks across the minor version this PR will be included in.